File: flutter_twilio_commkit_ios/ios/Classes/Core/TwilioLogger.swift
import Foundation
import os.log

/// Internal logger for the iOS plugin.
final class TwilioLogger {

    enum Level: Int, Comparable {
        case debug = 0, warning, error, none

        static func < (lhs: Level, rhs: Level) -> Bool { lhs.rawValue < rhs.rawValue }
    }

    private static var currentLevel: Level = .none
    private static let log = OSLog(subsystem: "com.twiliocommkit", category: "SDK")

    static func configure(level: String) {
        switch level.lowercased() {
        case "debug":   currentLevel = .debug
        case "warning": currentLevel = .warning
        case "error":   currentLevel = .error
        default:        currentLevel = .none
        }
    }

    static func debug(_ message: String) {
        guard currentLevel <= .debug else { return }
        os_log(.debug, log: log, "%{public}@", message)
    }

    static func warning(_ message: String) {
        guard currentLevel <= .warning else { return }
        os_log(.default, log: log, "%{public}@", message)
    }

    static func error(_ message: String) {
        guard currentLevel <= .error else { return }
        os_log(.error, log: log, "%{public}@", message)
    }
}

File: flutter_twilio_commkit_android/android/src/main/kotlin/com/twiliocommkit/android/core/TwilioLogger.kt
package com.twiliocommkit.android

/**
 * Internal logger for the Android plugin.
 */
object TwilioLogger {
    private var level: Level = Level.NONE

    enum class Level { DEBUG, WARNING, ERROR, NONE }

    fun configure(levelName: String) {
        level = when (levelName.lowercase()) {
            "debug" -> Level.DEBUG
            "warning" -> Level.WARNING
            "error" -> Level.ERROR
            else -> Level.NONE
        }
    }

    fun debug(message: String) {
        if (level <= Level.DEBUG) android.util.Log.d("TwilioCommKit", message)
    }

    fun warning(message: String) {
        if (level <= Level.WARNING) android.util.Log.w("TwilioCommKit", message)
    }

    fun error(message: String, throwable: Throwable? = null) {
        if (level <= Level.ERROR) android.util.Log.e("TwilioCommKit", message, throwable)
    }

    private operator fun Level.compareTo(other: Level): Int = ordinal - other.ordinal
}

